jest.polyfills.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @note The block below contains polyfills for Node.js globals
+ * required for Jest to function when running JSDOM tests.
+ * These HAVE to be require's and HAVE to be in this exact
+ * order, since "undici" depends on the "TextEncoder" global API.
+ *
+ * Consider migrating to a more modern test runner if
+ * you don't want to deal with this.
+ */
+
+const { TextDecoder, TextEncoder } = require('node:util');
+
+Object.defineProperties(globalThis, {
+  TextDecoder: { value: TextDecoder },
+  TextEncoder: { value: TextEncoder },
+});
+
+const { Blob } = require('node:buffer');
+
+const { fetch, Headers, FormData, Request, Response } = require('undici');
```

Comment:
Why are you using `undici`? Fetch API is also supported in nodejs starting from v18.

Reply:
I had an issue with it and found a solution at official [MSW docs](https://mswjs.io/docs/migrations/1.x-to-2.x#frequent-issues).

Reply:
Is it going to work if you replace **`undici`** with **`node:fetch`** or just **`fetch`**? Make sure that you use Node version >= 18. **20**.9.0 LTS is available.

Reply:
Not working with `node:fetch` or `fetch`. 
I am using the 18.18.2 Node version.